### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by charlielizzy**
we have completed tasks 3-8 and added some styling. the functionality and testing still works for the previous tasks. if you are happy with it we can merge into master.

### CODE REVIEW

File: src/components/comment.jsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    if (reply === '') {
+      return;
+    }
+
+    const replyData = {
+      id: 0,
+      content: reply,
+      createdAt: 'now',
+      score: 0,
+      replyingTo: '',
```
Comment (TomGroombridge) [inline src/components/comment.jsx]: Can we work out how to add this data and the below for the user?

File: src/components/comment.jsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  const [comments, setComments] = useState(props.comments);
+
+  const handleReplySubmit = (e) => {
+    e.preventDefault();
+    if (reply === '') {
+      return;
+    }
+
+    const replyData = {
+      id: 0,
```
Comment (TomGroombridge) [inline src/components/comment.jsx]: could you think about using this package to create a unique id? 

https://www.npmjs.com/package/uuid

File: src/components/comment.jsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    props.setComments(newComments);
+  };
+
+  return (
+    <div className="bg-white m-2 flex flex-row p-2" id="comment-container">
+      <div id="score-container" className="bg-gray-100 text-purple-900 m-2 rounded-lg h-[130px] w-[30px] flex flex-col justify-center">
+      
+        <button
+        id={`plus-button-${props.comment.id}`}
+        onClick={() => handlePlus()}
```
Comment (TomGroombridge) [inline src/components/comment.jsx]: if the function does not pass in an event or an argument you can just do 
```onClick={handlePlus}```

likewise for minus

File: src/components/comment.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        <Delete />
+        </div>
+      </div>
+      
+      <h1  id={`comment-${props.comment.id}`}>
+        {props.comment.content}
+      </h1>
+
+      <div className="flex p-2">
+        {/* <Edit id={props.comment.id} content={props.comment.content} commentUser={props.comment.user.username} comments={comments} setComments={props.setComments} currentUser={props.currentUser}/>
```
Comment (TomGroombridge) [inline src/components/comment.jsx]: reason for this code to be commented out?

File: src/components/delete.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+
+const Delete = () => {
+    const handleDelete = () => {
+        console.log("deleted")
+    }
+    
+return (
+    <div>
+        <button className="border-2 border-red-500 text-red-500 p-1 m-1 flex items-center" onClick={() => handleDelete()}><p>Delete</p><img src="/icons/icon-delete.svg"/></button>
```
Comment (TomGroombridge) [inline src/components/delete.jsx]: the formatting looks strange here, also can you call handleDelete without the anonymous function?

File: src/components/edit.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+const Edit = (props) => {
+    const [content, setContent] = useState(props.content);
+    const [editClicked, setEditClicked] = useState(false);
+    const [cantEdit, setCantEdit] = useState(false)
+
+
+    const handleSave = (e) => {
+e.preventDefault();
+if (content === '') {
```
Comment (TomGroombridge) [inline src/components/edit.jsx]: formatting

File: src/pages/Comments/index.jsx [hunk 7]
```diff
@@ -1,41 +1,91 @@
...
+    e.preventDefault();
+    if ( addedComment === '') {
+      return;
     }
-    catch (error) {
-      setLoading(false)
-      setError(true)
+
+    const addedCommentData = {
+        "id": 5,
```
Comment (TomGroombridge) [inline src/pages/Comments/index.jsx]: again can we add the correct user data in this object and maybe a uniqueId

File: src/pages/Comments/index.jsx [hunk 8]
```diff
@@ -1,41 +1,91 @@
...
+          "image": {
+            "png": "",
+            "webp": ""
+          },
+          "username": username,
+        },
+        "replies": []
     }
+  
+    let newComments = comments.concat(addedCommentData);
```
Comment (TomGroombridge) [inline src/pages/Comments/index.jsx]: this could be a const here i think? that might have been my fault

File: src/components/comment.jsx (hunk 1 above)
Comment (charlielizzy) [reply src/components/comment.jsx]: this has been updated now

File: src/components/comment.jsx (hunk 2 above)
Comment (ArgoNavis1763) [reply src/components/comment.jsx]: all done

File: src/components/comment.jsx (hunk 3 above)
Comment (charlielizzy) [reply src/components/comment.jsx]: done :)

File: src/components/comment.jsx (hunk 4 above)
Comment (ArgoNavis1763) [reply src/components/comment.jsx]: gone now

Comment (charlielizzy) [reply src/components/comment.jsx]: done

File: src/pages/Comments/index.jsx (hunk 7 above)
Comment (ArgoNavis1763) [reply src/pages/Comments/index.jsx]: unique ID added

File: src/pages/Comments/index.jsx (hunk 8 above)
Comment (ArgoNavis1763) [reply src/pages/Comments/index.jsx]: updated

**Pull request merged**